etc/dev-bin/defn-order-test.py

```diff
@@ -1,46 +1,48 @@
 #!/usr/bin/env python2
 
-import time, string, random
+import secrets
+import string
+import time
 from copy import deepcopy
 
 # This is a standalone performance test of the algorithm used in gcylc to
 # sort namespaces into "definition order", i.e. the order in which they are
 # defined in the suite.rc file.
 
 # Number of namespaces.
-N=10000
+N = 10000
 
 # N names of length 5-15 characters each (c.f. namespaces in "definition
 # order").
 names = []
-for i in range(0,N):
-    names.append( ''.join( random.choice(string.ascii_letters) for n in xrange( 5+random.randrange(10) )))
+for i in range(0, N):
+    names.append(''.join(secrets.choice(string.ascii_letters)
+                         for n in xrange(5+randrange(10))))
 
 # N lists with 2-7 names each (c.f. tree view paths of the inheritance
 # hierarchy).
 paths1 = []
-for i in range(0,N):
+for i in range(0, N):
     p = []
-    for j in range(0, 2+random.randrange(6)):
-        z = random.randrange(0,N)
-        p.append( names[z] )
+    for j in range(0, 2+secrets.randrange(6)):
+        z = secrets.randrange(0, N)
```

Comment:
This file is just an old POC for a particular sort-order problem, not used at all in the cylc library, which is why it's under `etc/dev` - so kind of a pointless change, but no matter.

Reply:
good to know ;-)